sql/catalog/lease: TestLeaseTxnDeadlineExtensionWithSession failed

sql/catalog/lease.TestLeaseTxnDeadlineExtensionWithSession [failed](https://tanzanite.cluster.engflow.com/invocations/default/91b9bbe9-e44f-4a24-b21e-7a37f30041ce?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [f2ce52ff7cc3b4680e0fec2ef844baad49042d6a](https://github.com/cockroachdb/cockroach/commits/f2ce52ff7cc3b4680e0fec2ef844baad49042d6a):


```
=== RUN   TestLeaseTxnDeadlineExtensionWithSession
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession3577010830
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:3417: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession3577010830
--- FAIL: TestLeaseTxnDeadlineExtensionWithSession (144.42s)
=== RUN   TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    lease_test.go:3413: 
        	Error Trace:	pkg/sql/catalog/lease/lease_test.go:3413
        	Error:      	Received unexpected error:
        	            	pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=576c66e1 key=/Tenant/10/Table/104/1/1024739012777672705/0 iso=Serializable pri=0.04912554 epo=0 ts=1732795925.647522933,2 min=1732795829.182709601,0 seq=1} lock=true stat=PENDING rts=1732795829.182709601,0 wto=false gul=1732795829.682709601,0
        	Test:       	TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    --- FAIL: TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked (99.63s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTxnDeadlineExtensionWithSession.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/lease.TestLeaseTxnDeadlineExtensionWithSession [failed](https://tanzanite.cluster.engflow.com/invocations/default/d28aec5f-4dce-4abc-b21c-f64fd62ebc6c?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [7d48198a57f014a8828194b90098699f70f0695a](https://github.com/cockroachdb/cockroach/commits/7d48198a57f014a8828194b90098699f70f0695a):


```
=== RUN   TestLeaseTxnDeadlineExtensionWithSession
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession4015649333
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:3417: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession4015649333
--- FAIL: TestLeaseTxnDeadlineExtensionWithSession (144.59s)
=== RUN   TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    lease_test.go:3413: 
        	Error Trace:	pkg/sql/catalog/lease/lease_test.go:3413
        	Error:      	Received unexpected error:
        	            	pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=66549340 key=/Tenant/10/Table/104/1/1025013925011062785/0 iso=Serializable pri=0.00409018 epo=0 ts=1732879822.249382301,2 min=1732879725.830408861,0 seq=1} lock=true stat=PENDING rts=1732879725.830408861,0 wto=false gul=1732879726.330408861,0
        	Test:       	TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    --- FAIL: TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked (99.54s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTxnDeadlineExtensionWithSession.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/lease.TestLeaseTxnDeadlineExtensionWithSession [failed](https://tanzanite.cluster.engflow.com/invocations/default/031e68d1-20c1-46b8-b00f-93e9aa573a8e?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [bcc993d796d03664604bf695e38fd5644d0bc952](https://github.com/cockroachdb/cockroach/commits/bcc993d796d03664604bf695e38fd5644d0bc952):


```
=== RUN   TestLeaseTxnDeadlineExtensionWithSession
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession206661211
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_test.go:3417: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession206661211
--- FAIL: TestLeaseTxnDeadlineExtensionWithSession (141.70s)
=== RUN   TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    lease_test.go:3413: 
        	Error Trace:	pkg/sql/catalog/lease/lease_test.go:3413
        	Error:      	Received unexpected error:
        	            	pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=e3dea909 key=/Table/104/1/1025298177725005825/0 iso=Serializable pri=0.00135247 epo=0 ts=1732966569.236390488,2 min=1732966473.902889077,0 seq=1} lock=true stat=PENDING rts=1732966473.902889077,0 wto=false gul=1732966474.402889077,0
        	Test:       	TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    --- FAIL: TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked (98.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTxnDeadlineExtensionWithSession.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/lease.TestLeaseTxnDeadlineExtensionWithSession [failed](https://tanzanite.cluster.engflow.com/invocations/default/b89582ed-f94a-4b03-804c-863887f2d977?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [b1474fa887606008960634b571cf4501efb6281b](https://github.com/cockroachdb/cockroach/commits/b1474fa887606008960634b571cf4501efb6281b):


```
=== RUN   TestLeaseTxnDeadlineExtensionWithSession
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession1732766126
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_test.go:3417: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession1732766126
--- FAIL: TestLeaseTxnDeadlineExtensionWithSession (141.73s)
=== RUN   TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    lease_test.go:3413: 
        	Error Trace:	pkg/sql/catalog/lease/lease_test.go:3413
        	Error:      	Received unexpected error:
        	            	pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=7e3c5727 key=/Table/104/1/1025594950279757825/0 iso=Serializable pri=0.07735286 epo=0 ts=1733057137.077574104,2 min=1733057041.576264107,0 seq=1} lock=true stat=PENDING rts=1733057041.576264107,0 wto=false gul=1733057042.076264107,0
        	Test:       	TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    --- FAIL: TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked (98.63s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTxnDeadlineExtensionWithSession.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/lease.TestLeaseTxnDeadlineExtensionWithSession [failed](https://tanzanite.cluster.engflow.com/invocations/default/d295ec38-e4a6-4b7c-bdc1-a09ee1be59f9?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [749b563a7b5957b2db487c6e8e2fdeed8630103d](https://github.com/cockroachdb/cockroach/commits/749b563a7b5957b2db487c6e8e2fdeed8630103d):


```
=== RUN   TestLeaseTxnDeadlineExtensionWithSession
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession1158554426
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:3417: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession1158554426
--- FAIL: TestLeaseTxnDeadlineExtensionWithSession (144.61s)
=== RUN   TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    lease_test.go:3413: 
        	Error Trace:	pkg/sql/catalog/lease/lease_test.go:3413
        	Error:      	Received unexpected error:
        	            	pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=b2eafcc6 key=/Tenant/10/Table/104/1/1025866103327522817/0 iso=Serializable pri=0.02313019 epo=0 ts=1733139886.476338466,2 min=1733139790.025365902,0 seq=1} lock=true stat=PENDING rts=1733139790.025365902,0 wto=false gul=1733139790.525365902,0
        	Test:       	TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    --- FAIL: TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked (99.53s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTxnDeadlineExtensionWithSession.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/lease.TestLeaseTxnDeadlineExtensionWithSession [failed](https://tanzanite.cluster.engflow.com/invocations/default/292a8be9-004e-4f10-ae7d-1d648235cd0a?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
=== RUN   TestLeaseTxnDeadlineExtensionWithSession
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession3020300840
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:3419: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession3020300840
--- FAIL: TestLeaseTxnDeadlineExtensionWithSession (147.79s)
=== RUN   TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    lease_test.go:3415: 
        	Error Trace:	pkg/sql/catalog/lease/lease_test.go:3415
        	Error:      	Received unexpected error:
        	            	pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=3192aa38 key=/Tenant/10/Table/104/1/1026162220451266561/0 iso=Serializable pri=0.07058573 epo=0 ts=1733230254.208210266,2 min=1733230157.873367993,0 seq=1} lock=true stat=PENDING rts=1733230157.873367993,0 wto=false gul=1733230158.373367993,0
        	Test:       	TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    --- FAIL: TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked (99.45s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTxnDeadlineExtensionWithSession.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/lease.TestLeaseTxnDeadlineExtensionWithSession [failed](https://tanzanite.cluster.engflow.com/invocations/default/4fdd7866-a9cc-43a6-bc6e-4fdf98dffced?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [44de2d379610067e14a7ebfbc92e64311f13a232](https://github.com/cockroachdb/cockroach/commits/44de2d379610067e14a7ebfbc92e64311f13a232):


```
=== RUN   TestLeaseTxnDeadlineExtensionWithSession
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession1487677691
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:3419: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseTxnDeadlineExtensionWithSession1487677691
--- FAIL: TestLeaseTxnDeadlineExtensionWithSession (144.94s)
=== RUN   TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    lease_test.go:3415: 
        	Error Trace:	pkg/sql/catalog/lease/lease_test.go:3415
        	Error:      	Received unexpected error:
        	            	pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=3e07b297 key=/Tenant/10/Table/104/1/1026437523867860993/0 iso=Serializable pri=0.01660101 epo=0 ts=1733314270.088338784,2 min=1733314173.953331307,0 seq=1} lock=true stat=PENDING rts=1733314173.953331307,0 wto=false gul=1733314174.453331307,0
        	Test:       	TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked
    --- FAIL: TestLeaseTxnDeadlineExtensionWithSession/validate-lease-txn-deadline-ext-blocked (99.31s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTxnDeadlineExtensionWithSession.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -3216,6 +3216,10 @@ SELECT * FROM T1`)
 func TestLeaseTxnDeadlineExtensionWithSession(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	// We may not see the desired transaction retry error when running under
+	// race since the txn could end up being aborted because of availability
+	// load in this configuration.
+	skip.UnderRace(t)
 
 	ctx := context.Background()
 	filterMu := syncutil.Mutex{}

